src/mono/mono/component/debugger-agent.c

```diff
@@ -1144,16 +1145,10 @@ socket_transport_connect (const char *address)
 		PRINT_DEBUG_MSG (1, "Listening on %s:%d (timeout=%d ms)...\n", host, port, agent_config.timeout);
 
 		if (agent_config.timeout) {
-			fd_set readfds;
-			struct timeval tv;
-
-			tv.tv_sec = 0;
-			tv.tv_usec = agent_config.timeout * 1000;
-			FD_ZERO (&readfds);
-			FD_SET (sfd, &readfds);
-
-			res = select ((int)sfd + 1, &readfds, NULL, NULL, &tv);
-
+			mono_pollfd mp;
+			mp.fd = (int)sfd;
+			mp.events = MONO_POLLIN;
+			res = mono_poll (&mp, 1, agent_config.timeout);
```

Comment:
We're not multiplying by 1k anymore? Inside of `mono_poll` we are dividing by 1k
https://github.com/dotnet/runtime/blob/3aec142ed1cb29e2a2f0588900347eee491b9c51/src/mono/mono/utils/mono-poll.c#L75
so maybe we should keep it upscaled?

Reply:
`agent_config.timeout` is in milliseconds.   `tv.tv_usec` was in microseconds.   `mono_poll` also takes a timeout in milliseconds.  There are actually several implementations.  The one where we're dividing is converting from milliseconds to seconds+microseconds.

tl;dr there's no need to do any conversion here at the call site for `mono_poll`